apps/veyfi/components/LockTab.tsx

```diff
@@ -0,0 +1,174 @@
+	const [increaseLockAmount, increaseLockAmountStatus] = useTransaction(VotingEscrowActions.increaseLockAmount, onTxSuccess);
+
+	const web3Provider = provider as ethers.providers.Web3Provider;
+	const userAddress = address as TAddress;
+	const hasLockedAmount = formatBN(positions?.deposit?.balance).gt(0);
+	
+	const unlockTime = useMemo((): TMilliseconds => {
+		return positions?.unlockTime || Date.now() + fromWeeks(toTime(lockTime));
+	}, [positions?.unlockTime, lockTime]);
+
+	const votingPower = useMemo((): BigNumber => {
+		return getVotingPower(formatBN(positions?.deposit?.underlyingBalance).add(lockAmount.raw),  unlockTime);
+	}, [positions?.deposit?.underlyingBalance, lockAmount, unlockTime]);
+
+	useEffect((): void => {
+		if(!positions?.unlockTime) {
+			return;
+		}
+		set_lockTime(toWeeks(getTimeUntil(positions.unlockTime), false).toString());
+	}, [positions?.unlockTime]);
+
+	const {isValid: isApproved} = validateAllowance({
+		tokenAddress: toAddress(votingEscrow?.token),
+		spenderAddress: toAddress(votingEscrow?.address),
+		allowances,
+		amount: lockAmount.raw
+	});
+	
+	const {isValid: isValidLockAmount, error: lockAmountError} = validateAmount({
+		amount: lockAmount.normalized,
+		balance: tokenBalance.normalized,
+		minAmountAllowed: hasLockedAmount ? 0 : MIN_LOCK_AMOUNT,
+		shouldDisplayMin: !hasLockedAmount
+	});
+	
+	const {isValid: isValidLockTime, error: lockTimeError} = validateAmount({
+		amount: lockTime,
+		minAmountAllowed: hasLockedAmount ? 0 : MIN_LOCK_TIME
+	});
+
+	const {isValid: isValidNetwork} = validateNetwork({supportedNetwork: 1, walletNetwork: chainID});
+	
+	const executeApprove = (): void => {
+		if (!votingEscrow || !userAddress) {
+			return;
+		}
+		approveLock(web3Provider, userAddress, votingEscrow.token, votingEscrow.address);
+	};
+	
+	const executeLock = (): void => {
+		if (!votingEscrow  || !userAddress) {
+			return;
+		}
+		lock(web3Provider, userAddress, votingEscrow.address, lockAmount.raw, toSeconds(unlockTime));
+	};
+	
+	const executeIncreaseLockAmount = (): void => {
+		if (!votingEscrow  || !userAddress) {
+			return;
+		}
+		increaseLockAmount(web3Provider, userAddress, votingEscrow.address, lockAmount.raw);
+	};
+
+	const txAction = !isApproved
+		? {
+			label: 'Approve',
+			onAction: executeApprove,
+			status: approveLockStatus.loading,
+			disabled: !isActive || !isValidNetwork || isApproved || isLoadingVotingEscrow
+		}
+		: hasLockedAmount
+			? {
+				label: 'Lock',
+				onAction: executeIncreaseLockAmount,
+				status: increaseLockAmountStatus.loading,
+				disabled: !isActive || !isValidNetwork || !isApproved || !isValidLockAmount || !isValidLockTime || isLoadingVotingEscrow
+			}
+			: {
+				label: 'Lock',
+				onAction: executeLock,
+				status: lockStatus.loading,
+				disabled: !isActive || !isValidNetwork || !isApproved || !isValidLockAmount || !isValidLockTime || isLoadingVotingEscrow
+			};
```

Comment:
nit: I find these nested ternary operations a bit hard to read, what you think if we refactor to something like:
```typescript
type TTxAction = {
	label: string;
	onAction: () => void;
	status?: boolean;
	disabled: boolean;
}

function getTxAction(): TTxAction {
	if (!isApproved) {
		const isDisabled = !isActive || !isValidNetwork || isApproved || isLoadingVotingEscrow;
		return {label: 'Approve', onAction: executeApprove, status: approveLockStatus.loading, disabled: isDisabled};
	}
	
	const isDisabled = !isActive || !isValidNetwork || !isApproved || !isValidLockAmount || !isValidLockTime || isLoadingVotingEscrow;

	if (hasLockedAmount) {
		return {label: 'Lock', onAction: executeIncreaseLockAmount, status: increaseLockAmountStatus.loading, disabled: isDisabled};
	}

	return {label: 'Lock', onAction: executeLock, status: lockStatus.loading, disabled: isDisabled};
}
```

Reply:
Fair enough, how about the disabled state, can we DRY it a bit?